### PR TITLE
Setup Resources/Tables

### PR DESCRIPTION
**Pull request by hashmaster3k**
Creates the following tables:

- [x] User
- [x] UserGardens
- [x] Gardens
- [x] Sensor
- [x] GardenHealth
- [x] Plant
- [x] GardenPlant

Notes: 

Made tests for validations, relationships and creation of instances. All tests passing.

### CODE REVIEW

File: app/models/garden_plant.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class GardenPlant < ApplicationRecord
+  validates :planted_date, presence: true
+
```
Comment (leahriffell) [inline app/models/garden_plant.rb:3]: @hashmaster3k nice! hopefully from the user's perspective, we can allow them to add multiple records at once for the same plant vs having to enter separately for each plant.

File: app/models/plant.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class Plant < ApplicationRecord
+  validates :image, :name, :species, :description, :light_requirements, :water_requirements, :when_to_plant, :harvest_time, :common_pests, presence: true
```
Comment (leahriffell) [inline app/models/plant.rb:2]: Thanks for adding these fields! If for some reason we can't get any of these from the Trefle API, we can easily remove.

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,6 @@
+class User < ApplicationRecord
+  validates :email, :provider, :token, presence: true
```
Comment (leahriffell) [inline app/models/user.rb:2]: Do we want to add refresh token? Or are we waiting to see which provider we're using?

File: coverage/.last_run.json [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (leahriffell) [inline coverage/.last_run.json:1]: We should probably remove all coverage files from github in gitignore

File: db/migrate/20201028202547_create_sensors.rb [hunk 5]
```diff
@@ -0,0 +1,11 @@
+class CreateSensors < ActiveRecord::Migration[5.2]
+  def change
+    create_table :sensors do |t|
+      t.references :garden, forign_key: true
```
Comment (leahriffell) [inline db/migrate/20201028202547_create_sensors.rb:4]: typo here on all foreign keys

File: db/migrate/20201028202019_create_user_gardens.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class CreateUserGardens < ActiveRecord::Migration[5.2]
+  def change
+    create_table :user_gardens do |t|
+      t.references :user, forign_key: true
```
Comment (leahriffell) [inline db/migrate/20201028202019_create_user_gardens.rb:4]: typo here on all foreign keys

File: db/migrate/20201028204720_create_garden_healths.rb [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class CreateGardenHealths < ActiveRecord::Migration[5.2]
+  def change
+    create_table :garden_healths do |t|
+      t.references :sensor, forign_key: true
```
Comment (leahriffell) [inline db/migrate/20201028204720_create_garden_healths.rb:4]: typo here on all foreign keys

File: spec/models/user_garden_spec.rb [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+RSpec.describe UserGarden, type: :model do
+  describe 'relationships' do
+    it { should belong_to :user}
+    it { should belong_to :garden }
+  end
+
+  describe 'creation' do
+    it 'can create user garden' do
+      user = User.create!(email: 'test@gmail.com', provider: 'google', token: '1234')
```
Comment (leahriffell) [inline spec/models/user_garden_spec.rb:11]: maybe we could make the provider an enum as well? I think storing as an integer is more efficient than as a string

File: app/models/user.rb (hunk 3 above)
Comment (hashmaster3k) [reply app/models/user.rb:2]: I was waiting, we can easily add validation for refresh token later.

File: coverage/.last_run.json (hunk 4 above)
Comment (hashmaster3k) [reply coverage/.last_run.json:1]: Good catch!

File: spec/models/user_garden_spec.rb (hunk 8 above)
Comment (hashmaster3k) [reply spec/models/user_garden_spec.rb:11]: Agreed. However, maybe for now we focus on one oAuth and later add more we can change to enum?

**Review by leahriffell — APPROVED** [review]

**Review by leahriffell — APPROVED** [review]

**Pull request merged**